code-guideline: Tag name should be a unique identifier

yerabolu opened the pull request:
Following are the changes to variable names that are matching with tag names (Rule 5.7 violations)

In kernel.h, event_type is matching with a tag name in lib/os/onoff.c. Added a _ prefix to event_type and
also to the macro argument names.

In userspace.c, *dyn_obj is matching with the tag name dyn_obj in the file itslef. Changed it to dyn

In device.h, device_mmio.h, init.h and init.c, changed the *device to dev. Except for one instance
in init.h.

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

ceolin commented on kernel/userspace.c:
```diff
@@ -255,26 +255,26 @@ static void thread_idx_free(uintptr_t tidx)
 struct z_object *z_dynamic_object_create(size_t size)
 {
-	struct dyn_obj *dyn_obj;
+	struct dyn_obj *dyn_obj_create;
```
this is looking more like a function name than a variable name. I think would be better just `obj` if it is possible.

yerabolu replied:
couldn't use obj as it was used for another pointer in the struct. thought this might create confusion.

yerabolu replied:
changed it to obj_thread

ceolin commented on kernel/userspace.c:
```diff
@@ -332,7 +332,7 @@ void *z_impl_k_object_alloc(enum k_objects otype)
 void k_object_free(void *obj)
 {
-	struct dyn_obj *dyn_obj;
+	struct dyn_obj *dyn_obj_free;
```
ditto

ceolin commented on kernel/userspace.c:
```diff
@@ -410,7 +410,7 @@ static void unref_check(struct z_object *ko, uintptr_t index)
 
 #ifdef CONFIG_DYNAMIC_OBJECTS
-	struct dyn_obj *dyn_obj =
+	struct dyn_obj *dyn_obj_unref =
```
same thing

ceolin reviewed (commented):
I just suggest changing some names that you put the verb in the end to not look as functions.

nashif commented:
would like to see each change go into its own commit to make it easier to review. Each commit should then explain why it needs to be changed.

nashif commented on subsys/net/lib/config/init.c:
```diff
@@ -485,9 +485,9 @@ int net_config_init_app(struct device *device, const char *app_info)
 
 #if defined(CONFIG_NET_CONFIG_AUTO_INIT)
-static int init_app(struct device *device)
+static int init_app(struct device *device_config)
```
here you use device_config, earlier you used device_entry and other names, maybe just use shorthand dev and be consistent. I do not see how device_config is better here, especially since we have a config member in the device structure.

yerabolu replied:
made the change to dev_def

ceolin commented:
> would like to see each change go into its own commit to make it easier to review. Each commit should then explain why it needs to be changed.

what you mean ? They are addressing exactly the same thing, that is not re-use a tag name, e.g: struct device *device, enum test test., ....
Split these changes in different commit makes no sense IMHO because they are addressing exactly the same thing.

nashif commented:
> Split these changes in different commit makes no sense IMHO because they are addressing exactly the same thing.

it does make a lot of sense and it is easier for everyone, even for the submitter. Just look at the commit message:

"In few files, changed the pointer variable to
another name
In few others, variables used in multiple files
are matching with tag name. So changed the
variable name to match with macro names in
which they are used."

It makes sense to have the commit message match the change and avoid something like "In few files" and "in few others"....

Also it helps establish context, i.e. I change this in that file and related to this is another change in some other file. If everything is in one commit, it is very difficult to review.

yerabolu committed:
code-guideline: Tag name should be a unique identifier

Following are the changes to variable names that are matching
with tag names (Rule 5.7 violations)

In kernel.h, event_type is matching with a tag name in
lib/os/onoff.c. Added a _ prefix to event_type and
also to the macro argument names.

In userspace.c, *dyn_obj is matching with the tag name
dyn_obj in the file itslef. Changed it to dyn

In device.h, device_mmio.h, init.h and init.c,
changed the *device to dev. Except for one change in
init.h

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

ceolin commented:
> > Split these changes in different commit makes no sense IMHO because they are addressing exactly the same thing.
> 
> it does make a lot of sense and it is easier for everyone, even for the submitter. Just look at the commit message:
> 
> "In few files, changed the pointer variable to
> another name
> In few others, variables used in multiple files
> are matching with tag name. So changed the
> variable name to match with macro names in
> which they are used."
> 
> It makes sense to have the commit message match the change and avoid something like "In few files" and "in few others"....
> 
> Also it helps establish context, i.e. I change this in that file and related to this is another change in some other file. If everything is in one commit, it is very difficult to review.

I think the problem is with the commit message and not with with being in one commit. For me it is crazy split just name changes into N commits, this will flood the tree ...

yerabolu commented:
> > Split these changes in different commit makes no sense IMHO because they are addressing exactly the same thing.
> 
> it does make a lot of sense and it is easier for everyone, even for the submitter. Just look at the commit message:
> 
> "In few files, changed the pointer variable to
> another name
> In few others, variables used in multiple files
> are matching with tag name. So changed the
> variable name to match with macro names in
> which they are used."
> 
> It makes sense to have the commit message match the change and avoid something like "In few files" and "in few others"....
> 
> Also it helps establish context, i.e. I change this in that file and related to this is another change in some other file. If everything is in one commit, it is very difficult to review.

Made changes to commit message.

yerabolu commented:
> > > Split these changes in different commit makes no sense IMHO because they are addressing exactly the same thing.
> > 
> > 
> > it does make a lot of sense and it is easier for everyone, even for the submitter. Just look at the commit message:
> > "In few files, changed the pointer variable to
> > another name
> > In few others, variables used in multiple files
> > are matching with tag name. So changed the
> > variable name to match with macro names in
> > which they are used."
> > It makes sense to have the commit message match the change and avoid something like "In few files" and "in few others"....
> > Also it helps establish context, i.e. I change this in that file and related to this is another change in some other file. If everything is in one commit, it is very difficult to review.
> 
> I think the problem is with the commit message and not with with being in one commit. For me it is crazy split just name changes into N commits, this will flood the tree ...

Made changes to commit message.

jenmwms commented:
Nit - please add a blank line before Signed-off-by for readability and consistency. example: https://github.com/zephyrproject-rtos/zephyr/pull/27725/commits/22733ac9487935cb3538aa31fa6087ea11e549ba

yerabolu commented:
> Nit - please add a blank line before Signed-off-by for readability and consistency. example: [22733ac](https://github.com/zephyrproject-rtos/zephyr/commit/22733ac9487935cb3538aa31fa6087ea11e549ba)

Done. Thank You!!

ceolin commented:
to be honest the commit message explaining file by file what you have rename is not necessary. Imagine that you had changed 100 files ? (What could happen if we don't limit it to the kernel) 
There is one reason, that is the code guideline rule that forbids re-use a tag name. That is it, the commit itself is not changing any logic if it is just changing variable names, you just need to be consistent.
Now if you are changing a struct or a enum name, I 'd say that they should be in different commits since they maybe be changing the API and must be reviewed more carefully.

jenmwms commented on include/device.h:
```diff
@@ -208,7 +208,7 @@ struct device {
 #ifdef CONFIG_DEVICE_POWER_MANAGEMENT
 	/** Power Management function */
-	int (*device_pm_control)(struct device *device, uint32_t command,
+	int (*device_pm_control)(struct device *dev_def, uint32_t command,
```
Can you please share the rationale for appending with _def? What does it stand for?

jenmwms replied:
fwiw - I remember seeing `struct device *dev` used frequently in gpio drivers, example: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/gpio/gpio_intel_apl.c

andrewboie replied:
just call it 'dev'

ceolin replied:
+1 I think I have suggested the same thing before as well.

yerabolu replied:
changed it to dev

nashif commented on include/kernel.h:
```diff
@@ -5037,21 +5037,21 @@ struct k_poll_event {
 	{ \
 	.poller = NULL, \
-	.type = event_type, \
+	.type = type_event, \
```
reversing the string here give it a completely new meaning, giving names is often very difficult, but changing them to something that changes the meaning or make it less obvious is worse than the issue you are trying to fix.

andrewboie replied:
for macro argument names I prefer to just prefix all of them with `_`.

so for example:

`#define K_POLL_EVENT_INITIALIZER(_event_type, _event_mode, _event_obj)`

yerabolu replied:
Added the changes.

tyrantlizard77 reviewed (commented):
I find it really hard to believe that "struct device" and "device" represent a collision.

and if it does, we have to change far more in the kernel than what this patch touches.

what specific MISRA rule is this?

not gonna lie I really hate this change and would prefer we didn't do this unless there are no other options

jenmwms commented:
> what specific MISRA rule is this?

5.7 if I recall correctly. I remember seeing it on the commit but must have fell off. @yerabolu  please update the message :)

andrewboie commented on include/sys/device_mmio.h:
```diff
@@ -204,18 +204,18 @@ struct z_device_mmio_rom {
  */
-#define DEVICE_MMIO_ROM_PTR(device) \
-	((struct z_device_mmio_rom *)((device)->config))
+#define DEVICE_MMIO_ROM_PTR(dev_def) \
```
just call it `dev`

andrewboie commented on kernel/userspace.c:
```diff
@@ -255,26 +255,26 @@ static void thread_idx_free(uintptr_t tidx)
 struct z_object *z_dynamic_object_create(size_t size)
 {
-	struct dyn_obj *dyn_obj;
+	struct dyn_obj *obj_thread;
```
`obj_thread` not a good name, it isn't a thread.

call it `dyn` maybe? it's just a local variable

tyrantlizard77 reviewed (commented):
I reviewed the MISRA rule and will go along with it.

Some naming suggestions inline.

andrewboie commented:
Are you going to go through and fix all the code in `subsys/`, `drivers/`, etc? many examples of `struct device *device`

yerabolu commented:
> Are you going to go through and fix all the code in `subsys/`, `drivers/`, etc? many examples of `struct device *device`

Yes, will submit separate a pull request to address the change from *device to dev.

nashif commented:
#ceolin 

> to be honest the commit message explaining file by file what you have rename is not necessary. Imagine that you had changed 100 files ? (What could happen if we don't limit it to the kernel)

that's not what I was asking for. The original patch was really confusing with multiple different names used for the same thing, ie.. device. If you want to change 50 structures in 30 different subsystems in 100 files and use a different name for each change and all in one commit, good luck with that.

nashif commented on include/init.h:
```diff
@@ -74,20 +74,20 @@ void z_sys_init_run_level(int32_t level);
  * @param init_fn Address to the init function of the entry.
  *
- * @param device A device driver instance pointer or NULL
+ * @param dev_instance A device driver instance pointer or NULL
```
why not dev or _device?

yerabolu replied:
Tried using dev and it is throwing an error as .dev is used in the macro. And to add a prefix, all the macro parameters need to be changed to maintain the consistency and was not sure if that's a good idea.

andrewboie replied:
I recommend renaming all the macro args to begin with a leading underscore, don't know if this is part of our coding style but it's the cleanest way to avoid collisions

nashif commented on kernel/userspace.c:
```diff
@@ -332,7 +332,7 @@ void *z_impl_k_object_alloc(enum k_objects otype)
 void k_object_free(void *obj)
 {
-	struct dyn_obj *dyn_obj;
+	struct dyn_obj *obj_dyn;
```
just dyn like you did before maybe?

yerabolu replied:
changed it

nashif commented on kernel/userspace.c:
```diff
@@ -410,7 +410,7 @@ static void unref_check(struct z_object *ko, uintptr_t index)
 
 #ifdef CONFIG_DYNAMIC_OBJECTS
-	struct dyn_obj *dyn_obj =
+	struct dyn_obj *obj_unref =
```
ditto

yerabolu replied:
sorry, missed these ones.

ceolin commented:
> @ceolin
> 
> > to be honest the commit message explaining file by file what you have rename is not necessary. Imagine that you had changed 100 files ? (What could happen if we don't limit it to the kernel)
> 
> that's not what I was asking for. The original patch was really confusing with multiple different names used for the same thing, ie.. device. If you want to change 50 structures in 30 different subsystems in 100 files and use a different name for each change and all in one commit, good luck with that.

Nope, the original patch was inconsistently wrong IMHO. I suggested to have those changes consistently changed and possibly automated. I had suggested that corner cases or "special" cases should be addressed in separated commits / prs. But changes in batch in one single commit.
